src/main/java/edu/ucsb/cs156/happiercows/controllers/CowDeathController.java

```diff
@@ -0,0 +1,91 @@
+package edu.ucsb.cs156.happiercows.controllers;
+
+import edu.ucsb.cs156.happiercows.errors.EntityNotFoundException;
+import edu.ucsb.cs156.happiercows.entities.CowDeath;
+import edu.ucsb.cs156.happiercows.entities.UserCommons;
+import edu.ucsb.cs156.happiercows.repositories.CommonsRepository;
+import edu.ucsb.cs156.happiercows.repositories.CowDeathRepository;
+import edu.ucsb.cs156.happiercows.repositories.UserCommonsRepository;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.LocalDateTime;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Api(description = "Cow Deaths")
+@RequestMapping("/api/cowdeath")
+@RestController
+@Slf4j
+
+public class CowDeathController extends ApiController {
+    @Autowired
+    CommonsRepository commonsRepository;
+
+    @Autowired
+    UserCommonsRepository userCommonsRepository;
+
+    @Autowired
+    CowDeathRepository cowDeathRepository;
+
+    @ApiOperation(value = "Create a new CowDeath as admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/admin/post")
+    public CowDeath postCowDeath_admin(
+            @ApiParam("commonsId") @RequestParam long commonsId,
+            @ApiParam("ZonedDateTime") @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime ZonedDateTime,
```

Comment:
In retrospect, it might simplify things to not have this be a parameter, but instead just use the same mechanism that the Job.java entity uses:

Add this to CowDeath.java:
```
@EntityListeners(AuditingEntityListener.class)
```

With imports: 
```
import javax.persistence.EntityListeners;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
```

Then change the entity from:

```
 private LocalDateTime ZonedDateTime;
```

to:

```
    @CreatedDate
    private ZonedDateTime createdAt;
```

By the way, `ZonedDateTime` was never supposed to be the name of the field in the CowDeath entity; that was an typo/oversight.  `ZonedDateTime` is actually the type that we should be using instead of `LocalDateTime`.

Reply:
Fixed